bundle/config/mutator/expand_glob_paths_test.go

```diff
@@ -0,0 +1,140 @@
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config"
+	"github.com/databricks/cli/bundle/config/paths"
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/databricks-sdk-go/service/compute"
+	"github.com/databricks/databricks-sdk-go/service/pipelines"
+	"github.com/stretchr/testify/require"
+)
+
+func touchEmptyFile(t *testing.T, path string) {
+	err := os.MkdirAll(filepath.Dir(path), 0700)
+	require.NoError(t, err)
+	f, err := os.Create(path)
+	require.NoError(t, err)
+	f.Close()
+}
+
+func TestExpandGlobPathsInPipelines(t *testing.T) {
+	dir := t.TempDir()
+
+	touchEmptyFile(t, filepath.Join(dir, "test1.ipynb"))
+	touchEmptyFile(t, filepath.Join(dir, "test/test2.ipynb"))
+	touchEmptyFile(t, filepath.Join(dir, "test/test3.ipynb"))
+	touchEmptyFile(t, filepath.Join(dir, "test1.jar"))
+	touchEmptyFile(t, filepath.Join(dir, "test/test2.jar"))
+	touchEmptyFile(t, filepath.Join(dir, "test/test3.jar"))
+	touchEmptyFile(t, filepath.Join(dir, "test1.py"))
+	touchEmptyFile(t, filepath.Join(dir, "test/test2.py"))
+	touchEmptyFile(t, filepath.Join(dir, "test/test3.py"))
+
+	b := &bundle.Bundle{
+		Config: config.Root{
+			Path: dir,
+			Resources: config.Resources{
+				Pipelines: map[string]*resources.Pipeline{
+					"pipeline": {
+						Paths: paths.Paths{
+							ConfigFilePath: filepath.Join(dir, "resource.yml"),
+						},
+						PipelineSpec: &pipelines.PipelineSpec{
+							Libraries: []pipelines.PipelineLibrary{
+								{
+									Notebook: &pipelines.NotebookLibrary{
+										Path: "./**/*.ipynb",
+									},
+								},
+								{
+									Jar: "./*.jar",
+								},
+								{
+									File: &pipelines.FileLibrary{
+										Path: "./**/*.py",
+									},
+								},
+								{
+									Maven: &compute.MavenLibrary{
+										Coordinates: "org.jsoup:jsoup:1.7.2",
+									},
+								},
+								{
+									Notebook: &pipelines.NotebookLibrary{
+										Path: "./test1.ipynb",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	m := ExpandPipelineGlobPaths()
+	err := bundle.Apply(context.Background(), b, m)
+	require.NoError(t, err)
+
+	libraries := b.Config.Resources.Pipelines["pipeline"].Libraries
+	require.Len(t, libraries, 7)
+
+	// Making sure glob patterns are expanded correctly
+	require.True(t, containsNotebook(libraries, filepath.Join("test", "test2.ipynb")))
+	require.True(t, containsNotebook(libraries, filepath.Join("test", "test3.ipynb")))
+	require.True(t, containsFile(libraries, filepath.Join("test", "test2.py")))
+	require.True(t, containsFile(libraries, filepath.Join("test", "test3.py")))
+
+	// Making sure exact file references work as well
+	require.True(t, containsNotebook(libraries, "test1.ipynb"))
+
+	// Making sure other libraries are not replaced
+	require.True(t, containsJar(libraries, "./*.jar"))
+	require.True(t, containsMaven(libraries, "org.jsoup:jsoup:1.7.2"))
```

Comment:
One more case; what about absolute paths.

It's valid to refer to `/some/static/path/lib.py` as well, but with globbing it might disappear.